public/javascripts/table.jsx

```diff
@@ -0,0 +1,82 @@
+var DeviceRow = React.createClass({
+  lastSeen : function(timestamp) {
+    var second = 1000;
+    var minute = 60 * second;
+    var hour = 60 * minute;
+    var day = 24 * hour;
+
+    var diff = Date.now() - timestamp;
+
+    if (diff <= 59 * second) return Math.floor(diff/second) + " seconds ago";
+    if (diff <= 59 * minute) return Math.floor(diff/minute) + " minutes ago";
+    if (diff <= 23 * hour) return Math.floor(diff/hour) + " hours ago";
+
+    return Math.floor(diff/day) + " days ago";
+  },
+
+  render: function () {
+    return (
+    <tr>
+      <td>{this.props.num}</td>
+      <td>{this.props.device.name}</td>
+      <td>{this.props.device.ip}</td>
+      <td>{this.props.device.mac_addr}</td>
+      <td>{this.props.device.vendor}</td>
+      <td>{this.lastSeen(this.props.device.times.end)}</td>
+    </tr>)
+  },
+});
+
+var DeviceTable = React.createClass({
+  getInitialState: function () {
+    return {
+      data : [],
+      isLoading: true
+    };
+  },
+  componentWillMount: function () {
+    $.get(this.props.url, function(data) {
+      this.setState({ data : data });
+      this.setState({ isLoading : false })
+    }.bind(this));
+  },
+
+  render: function () {
+    if (this.state.isLoading) return <Loading />
+    var rows = [];
+    var index = 0;
+    this.state.data.forEach(function(device) {
+      rows.push(<DeviceRow key={device.mac_addr} device={device} num={index}/>);
+      index++;
+    });
+    return (
+      <table className="table table-striped">
```

Comment:
Could this stuff be moved some where else to help make it easier to maintain? (separate code from strings)